tests/Unit/inc/Engine/Admin/Beacon/Beacon/insertScript.php

```diff
@@ -28,32 +27,29 @@ public function setUp() {
 		parent::setUp();
 
 		$this->options = Mockery::mock( Options_Data::class );
-		$this->beacon  = new Beacon( $this->options, $this->filesystem->getUrl( 'wp-content/plugins/wp-rocket/views/settings' ) );
+		$this->beacon  = Mockery::mock( Beacon::class . '[insert_script]', [
```

Comment:
Here you want `insert_script` to run normally. But you want to mock `generate` method. The way Mockery works is you are defining which method will be mocked, i.e. a partial mock. 

```suggestion
		$this->beacon  = Mockery::mock( Beacon::class . '[generate]', [
```

[Here's an example](https://github.com/wp-media/wp-rocket/blob/develop/tests/Unit/inc/Engine/CDN/RocketCDN/NoticesSubscriber/promoteRocketcdnNotice.php#L23-L30)

Why are mocking `generate()` method? It's a separate public method which is used as dependency within the `insert_script` method. With a unit test, we want to test only the target method/function, while mocking each dependency.